File: solutions/ruby/binary-search/1/binary_search.rb
class BinarySearch
  attr_accessor :list
  
  def initialize(list)
    @list = list
  end

  def search_for(val)
    left = 0
    right = list.length
    
    while left <= right
      i = ((left + right)/2).floor
      
      case
        when list[i].nil?
          return nil
        when list[i] > val
          right = i - 1
        when list[i] < val
          left = i + 1
        else
          return i
      end
    end
  end
end

File: solutions/javascript/binary-search/1/binary-search.js
//
// This is only a SKELETON file for the 'Binary Search' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const find = (list, value) => {
  let left = 0;
  let right = list.length;

  while (left < right) {
    let mid = Math.floor((left + right) / 2);

    const item = list[mid];

    if (item > value) {
      right = mid;
    } else if (item < value) {
      left = mid + 1;
    } else {
      return mid;
    }
  }

  throw new Error("Value not in array");
};
